### PR TITLE
updating to jekyll templating for clarity

### PR DESCRIPTION
**Pull request by stedn**
Large code reorg and cleanup but no change in functionality.

Jekyll works by using an html layout from `_layouts/` that contain html fragments from  `_Includes/`.  I just broke up the head, header, slider, footer and javascript into separate includes in the `default` layout.  Then the default layout has the `{{ content }}` section which drops in the content specific to each page.

You might notice in `_layouts/default.html` that I also drop in the `slider.html` include only on the index.html page using the `{% if page.name == "index.html" %}`. I did that just because the slider was supposed to be contained inside the "hero_area" class.

One thing I did notice was that the version of bootstrap that @bustardchad grabbed here had a tracking script for leostop.com (now seemingly defunct) I removed that version of bootstrap.js and replaced with the official 4.0.0 version.

### CODE REVIEW

**Comment by stedn** [comment]
I suppose one thing to know is how to run jekyll locally.  After you install jekyll (https://jekyllrb.com/docs/installation/) you can run `bundle exec jekyll serve` in the local directory and that will compile the templates and serve it locally at `127.0.0.1:4000`.

File: _config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+title: Isthmus Carshare
+email: isthmuscarshare@gmail.com
+description: >- # this means to ignore newlines until "baseurl:"
+  Write an awesome description for your new site here. You can edit this
```
Comment (willocho) [inline _config.yml:4]: Maybe we should put our description from the front page here?

An affordable, member-owned, and community-oriented alternative to private car ownership in Madison that will enhance mobility and transportation equity in our city, reduce congestion, and promote environmental stewardship

File: _includes/head.html [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    <title>Isthmus Carshare</title>
+    <!-- bootstrap core css -->
+    <link rel="stylesheet" type="text/css" href="assets/css/bootstrap.css" />
+    <!-- font awesome style -->
+    <link href="assets/css/font-awesome.min.css" rel="stylesheet" />
+    <!-- Custom styles for this template -->
+    <link href="assets/css/style.css" rel="stylesheet" />
+    <!-- responsive style -->
+    <link href="assets/css/responsive.css" rel="stylesheet" />
+    <style>
```
Comment (willocho) [inline _includes/head.html:21]: I made a n issue to move these style tags into a new CSS file
https://github.com/isthmuscarshare/isthmuscarshare/issues/7

**Review by willocho — APPROVED** [review]

File: _includes/head.html (hunk 2 above)
Comment (stedn) [reply _includes/head.html:21]: thank you!

File: _config.yml (hunk 1 above)
Comment (stedn) [reply _config.yml]: updated

**Comment by stedn** [comment]
@bustardchad I think I will merge this in so @willocho can work on firebase auth stuff sooner.  If you have questions or issues let me know and I can address them or revert later.

**Pull request merged**